numpy/core/tests/test_mem_overlap.py

```diff
@@ -98,6 +98,7 @@ def test_overlapping_assignments():
 def test_diophantine_fuzz():
     # Fuzz test the diophantine solver
     rng = np.random.RandomState(1234)
+    random_intp = rng.random_intp
```

Comment:
Nitpick: ```random_intp``` is on the closed interval, but ```randint``` is on the half-open interval.  Perhaps this is not so consequential since the tests still pass, but you are changing the function being called.